frontend/src/hooks/useMessagesFilter.ts

```diff
@@ -0,0 +1,46 @@
+import {
+    Dispatch,
+    SetStateAction,
+    useEffect,
+    useMemo,
+    useTransition
+} from "react";
+import { useSearchParams } from "react-router-dom";
+
+type Messages = { id: string; message: string; author: string }[];
+
+interface Props {
+    messages: Messages;
+    searchMode: string;
+    handleSetFilteredMessages: Dispatch<SetStateAction<Messages>>;
+}
+
+const useMessagesFilter = ({
+    messages,
+    searchMode,
+    handleSetFilteredMessages
+}: Props) => {
+    const [isPending, startTransition] = useTransition();
+    const [searchParams, setSearchParams] = useSearchParams();
+
+    const filterMessages = useMemo(() => {
+        return messages.filter(({ message, author }) => {
+            const filter = searchParams.get("messagesFilter") || "";
+            if (filter === "") {
```

Comment:
If user didn't specify filter option, u must have default filtering option

Reply:
When user hasn't type anything in search bar, he get all results, so it's default.